src/getinfo.sml

```diff
@@ -73,310 +71,215 @@ fun getSpan (f: item * E.env) =
-                  exp = fn (env, (k, span), acc) => add env (Exp (k, span)) span acc,
-                  sgn_item = fn (env, (k, span), acc) => add env (Sgn_item (k, span)) span acc,
-                  sgn = fn (env, (k, span), acc) => add env (Sgn (k, span)) span acc,
-                  str = fn (env, (k, span), acc) => add env (Str (k, span)) span acc,
-                  decl = fn (env, (k, span), acc) => add env (Decl (k, span)) span acc,
-                  bind = fn (env, binder) =>
-                            case binder of
-                                U.Decl.RelK x => E.pushKRel env x
-                              | U.Decl.RelC (x, k) => E.pushCRel env x k
-                              | U.Decl.NamedC (x, n, k, co) => E.pushCNamedAs env x n k co
-                              | U.Decl.RelE (x, c) => E.pushERel env x c
-                              | U.Decl.NamedE (x, c) => #1 (E.pushENamed env x c)
-                              | U.Decl.Str (x, n, sgn) => #1 (E.pushStrNamed env x sgn)
-                              | U.Decl.Sgn (x, n, sgn) => #1 (E.pushSgnNamed env x sgn)
-                }
-                env
-                { smallestgoodpart = NONE
-                , smallest = { item = Str (str, { file = fileName
-                                                , first = { line = 0, char = 0}
-                                                , last = { line = Option.getOpt (Int.maxInt, 99999), char = 0} })
-                             , span = { file = fileName
-                                      , first = { line = 0, char = 0}
-                                      , last = { line = Option.getOpt (Int.maxInt, 99999), char = 0} }
-                             , env = env }
-                }
-                ( L.DStr (Compiler.moduleOf "fileName", 0, (L.SgnError, ErrorMsg.dummySpan), (str, {file = fileName, first = ErrorMsg.dummyPos, last = ErrorMsg.dummyPos}))
-                , {file = fileName, first = ErrorMsg.dummyPos, last = ErrorMsg.dummyPos})
+fun filterSgiItems (items: Elab.sgn_item list) : foundInEnv list =
+    let
+        fun mapF item =
+            case item of
+                (Elab.SgiVal (name, _, c), _) => [FoundExp (name, c)]
+              | (Elab.SgiCon (name, _, k, _), _) => [FoundCon (name, k)]
+              | (Elab.SgiDatatype ds, loc) =>
+                List.concat (List.map (fn (dtx, i, _, cs) =>
+                                          FoundExp (dtx, (Elab.CNamed i, loc))
+                                          :: List.map (fn (x, i, _) => FoundExp (x, (Elab.CRel i, loc))) cs) ds)
+              | (Elab.SgiDatatypeImp (x, i, _, _, _, _, cs), loc) =>
+                FoundExp (x, (Elab.CNamed i, loc))
+                :: List.map (fn (x, i, _) => FoundExp (x, (Elab.CRel i, loc))) cs
+              | (Elab.SgiStr (_, name,  _, sgn), _) =>
+                [FoundStr (name, sgn)]
+              | (Elab.SgiSgn (name,  _, sgn), _) =>
+                [FoundStr (name, sgn)]
```

Comment:
Does it make sense that structures and signatures get the same entries here?  An `SgiSgn` is introducing a name for a signature, rather than a module that follows a signature.

Reply:
Aah right. I don't think an SgiSgn should lead to anything here then, since atm we're not completing signatures (can be added later ofc).